### PR TITLE
tickets/DM-29516

### PR DESCRIPTION
**Pull request by mareuter**
* Update code to new black version
* Pin conda builder version
* Update pre-commit hook versions
* Fix documentation build issue
* Update unit tests to new base class
* Fix internal simulation mode variable
* Disable test due to salobj issue
* Set version class variable.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+"""
+
+from documenteer.conf.pipelinespkg import *  # noqa
+import lsst.ts.dsm  # noqa
+
+project = "ts_dsm"
+html_theme_options["logotext"] = project  # noqa
+html_title = project
+html_short_title = project
+doxylink = {}  # Avoid warning: Could not find tag file _doxygen/doxygen.tag
```
Comment (r-owen) [inline doc/conf.py:12]: Please restore the two intersphinx mappings.

File: python/lsst/ts/dsm/dsm_csc.py [hunk 2]
```diff
@@ -30,9 +30,13 @@ class DSMCSC(salobj.BaseCsc):
 
     enable_cmdline_state = True
     valid_simulation_modes = (0, 1, 2)
+    version = __version__
```
Comment (r-owen) [inline python/lsst/ts/dsm/dsm_csc.py:33]: Please consider adding a note to the class doc string or a code comment here (just above line 32) that the simulation modes are documented in the user guide. (I looked for them in the class doc string first, and would have appreciated the pointer).

File: tests/test_csc.py [hunk 3]
```diff
@@ -145,12 +144,13 @@ async def test_lifecycle_behavior(self):
             )
             self.assertEqual(state.summaryState, salobj.State.STANDBY)
 
-    async def test_bad_simulation_mode(self):
-        """Test to ensure bad simulation modes raise
-        """
+    async def xtest_bad_simulation_mode(self):
```
Comment (r-owen) [inline tests/test_csc.py:147]: Please restore this test.

Also: it should not need to be asynchronous (though it's fine with me if you prefer to leave the "async" in).

**Review by r-owen — APPROVED** [review]
My main request is to restore one test that you disabled.

I have a few other minor suggestions for you to consider.
Also for future pull requests please make reformatting (especially automatic reformatting) a separate commit. That would make it easier to identify meaningful changes.

File: doc/conf.py (hunk 1 above)
Comment (mareuter) [reply doc/conf.py:12]: Done.

File: python/lsst/ts/dsm/dsm_csc.py (hunk 2 above)
Comment (mareuter) [reply python/lsst/ts/dsm/dsm_csc.py:33]: Done.

File: tests/test_csc.py (hunk 3 above)
Comment (mareuter) [reply tests/test_csc.py]: Done.

**Comment by mareuter** [comment]
I addressed all the comments. Sorry for the format mixing. Overzealous rebase on my part.

File: tests/test_csc.py (hunk 3 above)
Comment (mareuter) [reply tests/test_csc.py]: Done.

**Pull request merged**